daml-assistant/daml-helper/src/DA/Daml/Helper/Ledger.hs

```diff
@@ -216,3 +235,54 @@ runLedgerNavigator flags remainingArguments = do
           ]
         , ["  }"]
         ]
+
+----------------
+-- HTTP JSON API
+----------------
+
+newtype Method = Method
+  { unMethod :: BS.ByteString
+  } deriving IsString
+
+newtype Path = Path
+  { unPath :: BS.ByteString
+  } deriving IsString
+
+-- | Run a request against the HTTP JSON API.
+httpJsonRequest :: A.FromJSON a => LedgerArgs -> Method -> Path -> IO (Response a)
+httpJsonRequest LedgerArgs {sslConfigM,tokM,port,host} method path = do
+  when (isJust sslConfigM) $
+    fail "The HTTP JSON API doesn't support TLS requests, but a TLS flag was set."
+  resp <-
+    httpJSON $
+    setRequestPort port $
+    setRequestHost (BSC.pack host) $
+    setRequestMethod (unMethod method) $
+    setRequestPath (unPath path) $
+    setRequestHeader
+      "authorization"
+      [BSC.pack $ sanitizeToken tok | Just (L.Token tok) <- [tokM]]
+      defaultRequest
+  let status = getResponseStatusCode resp
+  unless (status == 200) $
+    fail $ "Request failed with error code " <> show status
```

Comment:
Is there some form of error message we get from the JSON APU we could print here in case of a failure? It might also be better to `putStrLn` all the information we have and call `exitFailure`.

Reply:
We now print the full status containing the error.